streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -256,8 +357,9 @@ public void run() {
     /**
      * Shutdown this stream thread.
      */
-    public void close() {
-        running.set(false);
+    public synchronized void close() {
+        log.info("{} Closing", logPrefix);
```

Comment:
It is a bit confusing to have `Closing` and `Shutting down` at the same time. Maybe change to "Informed to shut down"?